main.go

```diff
@@ -95,114 +96,165 @@ func getEtcdConfig() (etcd.Config, error) {
 	return config, nil
 }
 
-func getGlobalASN(api etcd.KeysAPI) (uint32, error) {
-	res, err := api.Get(context.Background(), fmt.Sprintf("%s/global/as_num", CALICO_BGP), nil)
+type Server struct {
+	t         tomb.Tomb
+	bgpServer *bgpserver.BgpServer
+	cli       *calicocli.Client
+	etcd      etcd.KeysAPI
+}
+
+func NewServer() (*Server, error) {
+	config, err := calicocli.LoadClientConfigFromEnvironment()
 	if err != nil {
-		return 0, err
+		return nil, err
```

Comment:
I added a log [here](https://github.com/projectcalico/calico-bgp-daemon/pull/14/files#diff-7ddfb3e035b42cd70649cc33393fe32cR700) and [here]().

Should we log before returning an error?